### DISCUSSION TITLE
will it work for shadow-root (closed) element?

### DISCUSSION THREAD

**Discussion by sseenivasan89** [sukgu/shadow-automation-selenium#51] [Q&A] [answered]
Hi,

I have shadow-root (closed) element to be interact, i have tried all the possibilities that shared here
but it's not working, just want to confirm it has support for (closed) element

![shadow](https://user-images.githubusercontent.com/44080743/170929780-b863292d-0a12-477d-a274-bf79bf6544da.png)

**Comment by sukgu** [comment, marked as answer]
Hi @sseenivasan89 , closed shadow DOMs are not meant to be accessed by the outside world. This is a design feature. So it's not automatable. 

From official docs https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM

```
open means that you can access the shadow DOM using JavaScript written in the main page context, for example using the [Element.shadowRoot](https://developer.mozilla.org/en-US/docs/Web/API/Element/shadowRoot) property:


If you attach a shadow root to a custom element with mode: closed set, you won't be able to access the shadow DOM from the outside myCustomElem.shadowRoot returns null. This is the case with built-in elements that contain shadow DOMs, such as <video>.
```
